src/EditorFeatures/Core/Implementation/Classification/SyntacticClassificationTaggerProvider.TagComputer.cs

```diff
@@ -389,11 +389,11 @@ private void AddClassifiedSpansForTokens(IEditorClassificationService classifica
                     span.Snapshot.AsText(), span.Span.ToTextSpan(), classifiedSpans, CancellationToken.None);
             }
 
-            private void OnDocumentActiveContextChanged(object sender, DocumentEventArgs args)
+            private void OnDocumentActiveContextChanged(object sender, DocumentActiveContextChangedEventArgs args)
             {
-                if (_workspace != null)
+                if (_workspace != null && _workspace == args.Workspace)
                 {
-                    ParseIfThisDocument(null, args.Document.Project.Solution, args.Document.Id);
+                    ParseIfThisDocument(null, args.Workspace.CurrentSolution, args.NewContextDocumentId);
```

Comment:
The change in active document is unrelated to the state of the solution.  I don't see a need to pinpoint the exact snapshot that was current at the moment the active context changed.